lib/liquid/context.rb

```diff
@@ -82,7 +82,11 @@ def handle_error(e, line_number = nil)
       output = nil
 
       if exception_handler
-        result = exception_handler.call(e)
+        exception_handler_arity = exception_handler.arity
+        args = [e]
+        args << line_number if exception_handler_arity > 1
+        args << line if exception_handler_arity > 2
+        result = exception_handler.call(*args)
```

Comment:
Ah, excellent idea, saves us if we want to add more extra data later, than having to mess around with an additional arg, will do.